### PR TITLE
Platform/Messaging: Http endpoint

### PR DESCRIPTION
**Pull request by ILyaIL83**
Just added projects for Api/Implementation/UnitTests.

### CODE REVIEW

File: Source/NWheels.Platform.Messaging.Implementation/NWheels.Platform.Messaging.Implementation.csproj [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore" Version="1.1.1" />
+    <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="1.1.1" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\NWheels.Api\NWheels.Api.csproj" />
+    <ProjectReference Include="..\NWheels.Implementation\NWheels.Implementation.csproj" />
```
Comment (felix-b) [inline Source/NWheels.Platform.Messaging.Implementation/NWheels.Platform.Messaging.Implementation.csproj]: Do we really need a reference to NWheels.Implementation?

Comment (ILyaIL83) [inline Source/NWheels.Platform.Messaging.Implementation/NWheels.Platform.Messaging.Implementation.csproj]: Moved LifecycleListenerComponentBase to NWheels.Api from NWheels.Implementation.

**Comment by ILyaIL83** [comment]
Notes:
1. Haven't found a way to start Kestrel without dependencies to AspNetCore in NWheels.Host
2. I'm going to add RestApiFeatureLoader where I want to register RestApiService by IRestApiService for constructor injection in KestrelLifecycleListenerComponent
3. Tight coupling between Platform.Messaging and Platform.Rest
4. In RestApiMiddleware HttpRequestMessage of Net should be created from HttpRequest of AspNetCore and vice versa HttpResponse from HttpResponseMessage - feels strange.

**Comment by felix-b** [comment]
@ILyaIL83, per your notes:

1 - Ok for first happy path, needs to be resolved later on, please add a backlog task for that.

2+3 - ok for now, please add a nacklog task to resolva that later.

4 - see http://stackoverflow.com/questions/42155768/how-to-set-httpresponse-to-httpresponsemessage-in-web-api, does it answer the issue?

**Comment by ILyaIL83** [comment]
1. https://github.com/felix-b/NWheels/issues/23
2+3. https://github.com/felix-b/NWheels/issues/24
4. Technically there is no problem, but it smells(additional iteration/conversion)

**Comment by ILyaIL83** [comment]
I think it's ready for merge now.

**Comment by felix-b** [comment]
So we can now run a web server! Great job.

**Pull request merged**